effectful/handlers/torch.py

```diff
@@ -180,20 +170,70 @@ def reindex_flat_tensor(t):
             return t
 
         result = t.reshape(inds[0].shape + t.shape[1:])
-        return torch_getitem(result, tuple(var() for (var, _) in reindex_fvs))
+        return torch_getitem(result, tuple(k() for k in sized_fvs.keys()))
+
+    result = tree.map_structure(reindex_flat_tensor, flat_result)
+    return result
+
+
+@defop
+def to_tensor_(
+    t: Annotated[torch.Tensor, Scoped[A | B]],
+    *args: Annotated[Operation[[], torch.Tensor], Scoped[A]],
+) -> Annotated[torch.Tensor, Scoped[B]]:
+    def _evaluate(expr):
```

Comment:
Why is there a duplicate of `evaluate` here? This seems to suggest we should be changing the interface of `_partial_eval` to be an `apply` rule. Also, should `_partial_eval` just be absorbed into the body of `to_tensor_`? When would it ever be correct to use `_partial_eval` on its own?

Reply:
I had a bad time trying to overload `apply` without breaking something else, so I gave up. 

`_partial_eval` has other callers, so I left it as its own function.

Reply:
I only see one other caller (`_register_torch_op`). Should that be made to use `to_tensor_` instead, or should `_partial_eval` be absorbed into its body? Right now it seems like the logic is dispersed across a few different locations in a way that makes it hard to unwind.

Reply:
I looked at this a bit over the weekend, will put up a separate PR.